R/get_code.R

```diff
@@ -92,6 +94,46 @@ get_code.TealDataAbstract <- function(x, dataname = character(0), deparse = TRUE
   }
 }
 
+#' @rdname get_code
+#' @export
+#' @examples
+#'
+#' tdata1 <- teal_data()
+#' tdata1 <- within(tdata1, {
+#'   a <- 1
+#'   b <- a^5
+#'   c <- list(x = 2)
+#' })
+#' get_code(tdata1)
+#' get_code(tdata1, "a")
+#' get_code(tdata1, "b")
+#'
+#' tdata2 <- teal_data(x1 = iris, code = "x1 = iris")
+#' get_code(tdata2)
+#' get_code(verify(tdata2))
+#'
+get_code.teal_data <- function(x, names = NULL, deparse = TRUE, ...) {
+  check_ellipsis(...)
+  checkmate::assert_character(names, null.ok = TRUE)
+  checkmate::assert_flag(deparse)
+
+  code <- if (length(names) > 0) {
```

Comment:
```suggestion
  code <- if (!is.null(names)) {
```

Reply:
i'm not that convinced this is neccessary after we introduced a check for character to have min.len = 1

Reply:
But with min.len the only way to have something of length 0 here is to pass NULL.

Reply:
and NULL has lenght 0. thats why I left it. But I guess we use length() statements for objects of length 0 and NULLs so I'll accept your propsition